### PR TITLE
add resource variable provider

### PR DESCRIPTION
**Pull request by Estavaringo**
Add a resource variable provider to enable the access to resources through variables

### CODE REVIEW

File: src/Take.Blip.Builder/Variables/ResourceVariableProvider.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            }
+            catch (LimeException ex) when (ex.Reason.Code == ReasonCodes.COMMAND_RESOURCE_NOT_FOUND)
+            {
+                return null;
+            }
+        }
+
+        private async Task<Command> ExecuteGetResourceCommandAsync(string name, CancellationToken cancellationToken)
+        {
+            var getResourceCommand = new Command()
```
Comment (ceifa) [inline src/Take.Blip.Builder/Variables/ResourceVariableProvider.cs]: Use IResourceExtension instead

File: src/Take.Blip.Builder/Variables/ResourceVariableProvider.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Lime.Protocol;
+using Lime.Protocol.Network;
+using Lime.Protocol.Serialization;
+using Take.Blip.Client;
+
+namespace Take.Blip.Builder.Variables
+{
+    public class ResourceVariableProvider : IVariableProvider
```
Comment (andreminelli) [inline src/Take.Blip.Builder/Variables/ResourceVariableProvider.cs]: This implementation is almost the same as BucketVariableProvider. 
Extract a base class for them (or reuse `BucketVariableProvider` in the same idea as made by the `ResourceExtension`)

Comment (Estavaringo) [reply src/Take.Blip.Builder/Variables/ResourceVariableProvider.cs]: I created a base class for this implementation

File: src/Take.Blip.Builder.UnitTests/ContextBaseTests.cs [hunk 3]
```diff
@@ -415,6 +415,58 @@ public async Task GetBucketJsonVariablePropertyShouldSucceed()
...
+        }
+        [Fact]
+        public async Task GetResourceTextVariableShouldSucceed()
+        {
+            // Arrange
+            var document = new PlainText()
+            {
+                Text = "my value"
+            };
+            Sender
```
[the remark this answers is not in the document]
Comment (Estavaringo) [reply src/Take.Blip.Builder.UnitTests/ContextBaseTests.cs]: this setup is inside a method now

File: src/Take.Blip.Builder/Variables/BaseResourceVariableProvider.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Client;
+
+namespace Take.Blip.Builder.Variables
+{
+    public class BaseResourceVariableProvider : IVariableProvider
```
Comment (andreminelli) [inline src/Take.Blip.Builder/Variables/BaseResourceVariableProvider.cs]: Rename to `ResourceVariableProviderBase` to keep the same pattern used on this repo and make it `abstract`.

File: src/Take.Blip.Builder/Variables/BaseResourceVariableProvider.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        private readonly ISender _sender;
+        private readonly IDocumentSerializer _documentSerializer;
+        private readonly string _resourceName;
+        public VariableSource Source => throw new NotImplementedException();
```
Comment (andreminelli) [inline src/Take.Blip.Builder/Variables/BaseResourceVariableProvider.cs]: Make this property abstract, since inherited classes must override it.

**Pull request merged**